src/module/hooks/combat.mjs

```diff
@@ -270,3 +283,30 @@ async function renderOngoingEffectsCard(title, combatant, effectData) {
     };
     ChatMessage.create(chatData, {});
 }
+
+async function executeLifecycleMacro(combatant, hookName) {
+    // If this isn't the actor's player, emit a socket request for that player to execute the hook
+    if (game.user?.character?.id !== combatant.actor.id) {
+        return game.socket.emit('system.archmage', {
+            type: 'actorLifecycleHook',
+            actorId: combatant.actor.id,
+            hookName
+        });
+    }
+
+    const hookBody = combatant?.actor?.system?.lifecycleHooks?.[hookName]?.trim();
+    if (!hookBody) return;
+
+    // Can't run if you can't run
+    if (!game.user.hasPermission("MACRO_SCRIPT")) return;
+
+    // Run our own function to bypass macro parameters limitations - based on Foundry's _executeScript
+    const AsyncFunction = async function () {}.constructor;
+    try {
+        const fn = new AsyncFunction(hookBody);
+        await fn.call(combatant.actor);
```

Comment:
There are a few additional variables we provide to make embedded macros on items functionally similarly to standard macros, so I would incorporate those here as well. I'm not sure of exactly where that needs to happen, though the code for embedded macros should have some pointers. The variables I have noted in the tooltip were:
- `speaker`
- `actor`
- `token`
- `archmage` (not equivalent to `game.archmage` but is instead a collection of various convenience properties/methods

Reply:
Updated this to call with `this, speaker, actor, archmage`, but that last one is empty for now. Maybe leave it there for now and see what people ask for later on?